pandas/core/groupby/generic.py

```diff
@@ -1676,7 +1676,8 @@ def _wrap_transformed_output(
 
     def _wrap_agged_manager(self, mgr: Manager2D) -> DataFrame:
         if not self.as_index:
-            index = Index(range(mgr.shape[1]))
+            rows = mgr.shape[1] if mgr.shape[0] > 0 else 0
```

Comment:
comment here about why, pointing back to the issue/pr?

Reply:
Thanks - added.

Reply:
we should strive to move this code inside internal (separate PR for sure)